File: LeetCode Solutions/19. Remove Nth Node From End of List/solution19.js
/**
* Definition for singly-linked list.
* function ListNode(val, next) {
* this.val = (val===undefined ? 0 : val)
* this.next = (next===undefined ? null : next)
* }
*/

/**
* @param {ListNode} head
* @param {number} n
* @return {ListNode}
*/

var removeNthFromEnd = function(head, n) {
	let root = new ListNode(-1, head), tail = root
	while (n > 1) {
		head = head.next
		n--
	}
	while (head.next) {
		tail = tail.next
		head = head.next
	}
	tail.next = tail.next.next
	return root.next
};

File: LeetCode Solutions/19. Remove Nth Node From End of List/solution19.c
/**
* Definition for singly-linked list.
* struct ListNode {
* int val;
* struct ListNode *next;
* };
*/

struct ListNode* removeNthFromEnd(struct ListNode* head, int n){
	struct ListNode* root = (struct ListNode*) malloc(sizeof(struct ListNode));
	root->val = 0;
	root->next = head;
	struct ListNode* tail = root;
	while (n > 1) {
		head = head->next;
		n--;
	}
	while (head -> next != NULL) {
		head = head -> next;
		tail = tail -> next;
	}
	tail -> next = tail -> next -> next;
	return root -> next;
}